Find related containers: new functionality

kelby-shelton opened the pull request:
- Added support for specifying a field list to match against other containers

kelby-shelton committed:
Find related containers: new functionality

- Added support for specifying a field list to match against other containers

kelby-shelton committed:
Updated description

kelby-shelton committed:
Added earliest_time to find_common_container_by_field

- Added earliest_time to the function find_common_container_by_field in order to speed up artifact processing time.
- Adjusted debug to provide better feedback on successful/failed runs

philroyer-phantom commented on custom_functions/find_related_containers.py:
```diff
@@ -1,26 +1,28 @@
     Args:
-        value_list (CEF type: *): An indicator value to search on, such as a file hash or IP address. To search on all indicator values in the container, use "*".
-        minimum_match_count (CEF type: *): The minimum number of values from the value_list parameter that must match with related containers. Supports an integer or the string 'all'. Adding 'all' will set the minimum_match_count to the length of the number of unique values in the value_list. If no match count provided, this will default to 1.
+        field_list (CEF type: *): A field  to search on, such as destinationAddress or risk_object. Not compatible with value_list. Only containers that share the exact value in the exact fields will be considered related.
```
Can we add support for a comma-separated list of fields? The param is called field_list (which is plural), but then in the description it says "a field to search on" (which is singular).

The last sentence should be updated to reflect how minimum_match_count works. So maybe something like this: `Only fields that share the exact value and the exact field name will contribute to the match count`.

philroyer-phantom commented on custom_functions/find_related_containers.py:
```diff
@@ -1,26 +1,28 @@
-        value_list (CEF type: *): An indicator value to search on, such as a file hash or IP address. To search on all indicator values in the container, use "*".
-        minimum_match_count (CEF type: *): The minimum number of values from the value_list parameter that must match with related containers. Supports an integer or the string 'all'. Adding 'all' will set the minimum_match_count to the length of the number of unique values in the value_list. If no match count provided, this will default to 1.
+        field_list (CEF type: *): A field  to search on, such as destinationAddress or risk_object. Not compatible with value_list. Only containers that share the exact value in the exact fields will be considered related.
+        value_list (CEF type: *): An indicator value to search on, such as a file hash or IP address. To search on all indicator values in the container, use "*". Not compatible with field_list.
```
Again, value_list is plural but `and indicator value to search` is singular.

philroyer-phantom commented on custom_functions/find_related_containers.py:
```diff
@@ -1,26 +1,28 @@
-        filter_label: Optional comma-separated list of labels to filter on. Only containers that have labels matching an item in this list will be included.
+        filter_label (CEF type: phantom container label): Optional comma-separated list of labels to filter on. Only containers that have labels matching an item in this list will be included.
         filter_severity: Optional comma-separated list of severities to filter on. Only containers that have severities matching an item in this list will be included.
         filter_in_case: Optional parameter to filter containers that are in a case or not.
```
For filter_in_case, is it True returns only ones in cases, False returns only ones not in cases, and None returns all? If that is correct, it could say:

```Optional parameter. True for only containers in cases, False for only containers not in cases. Default is all containers.``` 

Or something very explicit like that.

philroyer-phantom commented on custom_functions/find_related_containers.json:
```diff
@@ -1,23 +1,30 @@
-            "name": "value_list",
-            "placeholder": "*"
+            "name": "field_list",
+            "placeholder": ""
```
I think this one could really use a placeholder. Just to clear up that it is a cef field name, not a full cef data path.

philroyer-phantom replied:
Maybe it should say "cef field" as well.

philroyer-phantom commented:
On line 10 I think the description needs to have `mon` for month instead of `m`

philroyer-phantom commented:
On line 10 I think the description needs to have `mon` for month instead of `m`

philroyer-phantom reviewed (changes requested):
Mostly minor changes requested but the comma-separated list for field list is more of a medium change.

kelby-shelton committed:
Changes to descriptions and added support for multivalued field_list

- Updated descriptions for field_list, value_list, and filter_in_case
- Updated placeholder values for field_list and value_list
- Changed python code to add support for comma separated field_list
- Adjusted related_artifacts check to not error out if 'data' key doesn't exist.

philroyer-phantom commented on custom_functions/find_related_containers.json:
```diff
@@ -8,19 +8,19 @@
             "input_type": "list",
             "name": "field_list",
-            "placeholder": ""
+            "placeholder": "destionationAddress, requestURL"
```
Minor typo. Should be destinationAddress
